ProcessOptimizer/space/space.py

```diff
@@ -1,917 +1,940 @@
+            else:
+                columns.append(dim.inverse_transform(Xt[:, start : start + offset]))
+
+            start += offset
+
+        # Transpose
+        rows = []
+
+        for i in range(len(Xt)):
+            r = []
+            for j in range(self.n_dims):
+                r.append(columns[j][i])
+
+            rows.append(r)
+
+        return rows
+
+    @property
+    def n_dims(self):
+        """The dimensionality of the original space."""
+        return len(self.dimensions)
+
+    @property
+    def transformed_n_dims(self):
+        """The dimensionality of the warped space."""
+        return sum([dim.transformed_size for dim in self.dimensions])
+
+    @property
+    def bounds(self):
+        """The dimension bounds, in the original space."""
+        b = []
+
+        for dim in self.dimensions:
+            if dim.size == 1:
+                b.append(dim.bounds)
+            else:
+                b.extend(dim.bounds)
+
+        return b
+
+    @property
+    def names(self):
+        """The names of the dimensions if given. Otherwise [X1, X2, ... Xn]"""
+        labels = [
+            "$X_{%i}$" % i if d.name is None else d.name
+            for i, d in enumerate(self.dimensions)
+        ]
+        return labels
+
+    def __contains__(self, point):
+        """Check that `point` is within the bounds of the space."""
+        for component, dim in zip(point, self.dimensions):
+            if component not in dim:
+                return False
+        return True
+
+    @property
+    def transformed_bounds(self):
+        """The dimension bounds, in the warped space."""
+        b = []
+
+        for dim in self.dimensions:
+            if dim.transformed_size == 1:
+                b.append(dim.transformed_bounds)
+            else:
+                b.extend(dim.transformed_bounds)
+
+        return b
+
+    @property
+    def is_categorical(self):
+        """Space contains exclusively categorical dimensions"""
+        return all([isinstance(dim, Categorical) for dim in self.dimensions])
+
+    @property
+    def is_partly_categorical(self):
+        """Space contains any categorical dimensions"""
+        return any([isinstance(dim, Categorical) for dim in self.dimensions])
+
+    def distance(self, point_a, point_b):
+        """Compute the L1 (Manhattan or taxicab) distance between two points in this space.
+
+        Parameters
+        ----------
+        * `a` [array]
+            First point.
+
+        * `b` [array]
+            Second point.
+        """
+        distance = 0.0
+        if len(self.dimensions) > 1:
+            for a, b, dim in zip(point_a, point_b, self.dimensions):
+                distance += dim.distance(a, b)
+
+        if len(self.dimensions) == 1:
+            distance += self.dimensions[0].distance(point_a[0], point_b[0])
+
+        return distance
+
+    def lhs(
+        self,
+        n: int,
+        random_state: Union[
```

Comment:
```suggestion
        seed: Union[
```
To encourage use of seeding rather than RandomState.